component/src/main/java/org/cbioportal/genome_nexus/component/annotation/CanonicalTranscriptResolver.java

```diff
@@ -52,4 +58,49 @@ else if (transcripts.size() > 1) {
                 variantAnnotation.getTranscriptConsequences(), variantAnnotation.getMostSevereConsequence());
         }
     }
+
+    public void setPredefinedCanonicalTranscriptIds(Set<String> predefinedCanonicalTranscriptIds) {
```

Comment:
Alternatively, we can add `persistence` module as a dependency to `component` module, and directly fetch genome nexus canonical transcript ids here.